src/data/PredictorD2ApiRepository.ts

```diff
@@ -58,6 +64,26 @@ export class PredictorD2ApiRepository implements PredictorRepository {
         return objects.map(({ id, displayName }) => ({ id, name: displayName }));
     }
 
+    public async getDataElements(): Promise<NamedRef[]> {
+        const outputObjects = (
+            await this.api.models.predictors
+                .get({ paging: false, fields: { output: true } })
+                .getData()
+        ).objects.map(element => element.output.id);
+
+        const dataElements = (
+            await this.api.models.dataElements
+                .get({
+                    filter: { id: { in: outputObjects } },
+                    paging: false,
+                    fields: { id: true, displayName: true },
+                })
+                .getData()
+        ).objects;
+
+        return dataElements.map(({ id, displayName }) => ({ id, name: displayName }));
+    }
```

Comment:
This can be simplified into one single API call:

```ts
        const { objects } = await this.api.models.predictors
            .get({ paging: false, fields: { output: { id: true, displayName: true } } })
            .getData();

        return objects.map(({ output }) => ({ id: output.id, name: output.displayName }));
```